src/nbl/asset/utils/CSmoothNormalGenerator.cpp

```diff
@@ -214,59 +270,59 @@ void CSmoothNormalGenerator::processConnectedVertices(asset::ICPUMeshBuffer * bu
 				{
 					if (processedVertex != bounds.begin)
 						if (compareVertexPosition(processedVertex->position, bounds.begin->position, epsilon) &&
-							vxcmp(*processedVertex, *bounds.begin, buffer))
+							vxcmp(*processedVertex, *bounds.begin, polygon))
 						{
 							//TODO: better mean calculation algorithm
 							normal += bounds.begin->parentTriangleFaceNormal * bounds.begin->wage;
 						}
 				}
 			}
-
-
-			normal = core::normalize(core::vectorSIMDf(normal));
-			buffer->setAttribute(normal, normalAttrID, buffer->getIndexValue(processedVertex->indexOffset));
+			normal = normalize(normal);
+		  memcpy(normalPtr + (normalStride * processedVertex->index), &normal, sizeof(normal));
 		}
 	}
 
+	CPolygonGeometryManipulator::recomputeContentHashes(outPolygon.get());
 
+	return outPolygon;
 }
 
-std::array<uint32_t, 8> CSmoothNormalGenerator::VertexHashMap::getNeighboringCellHashes(const IMeshManipulator::SSNGVertexData & vertex)
+std::array<uint32_t, 8> CSmoothNormalGenerator::VertexHashMap::getNeighboringCellHashes(const CPolygonGeometryManipulator::SSNGVertexData & vertex)
 {
 	std::array<uint32_t, 8> neighbourhood;
 
-	core::vectorSIMDf cellFloatCoord = vertex.position / cellSize - core::vectorSIMDf(0.5f);
-	core::vector3du32_SIMD neighbor = core::vector3du32_SIMD(static_cast<uint32_t>(cellFloatCoord.x), static_cast<uint32_t>(cellFloatCoord.y), static_cast<uint32_t>(cellFloatCoord.z));
+	hlsl::float32_t3 cellFloatCoord = vertex.position / m_cellSize - hlsl::float32_t3(0.5f);
+	hlsl::uint32_t3 neighbor = hlsl::uint32_t3(static_cast<uint32_t>(cellFloatCoord.x), static_cast<uint32_t>(cellFloatCoord.y), static_cast<uint32_t>(cellFloatCoord.z));
```

Comment:
`hlsl::_static_cast<hlsl::uint32_t3>(cellFloatCoord.x)` should work,

Btw there might be a problem that float->int conversions round towards zero, so [0.5,1.5) maps to 0 but also does (-0.5,0.5). Its probably not a great deal and doesn't impact correctness of the algo.